app/scraping.py

```diff
@@ -1,19 +1,40 @@
 from bs4 import BeautifulSoup
 import os
+import requests
+from PIL import Image
+from io import BytesIO
 
+# Define global variables for minimum image dimensions
+MIN_IMAGE_WIDTH = 100
+MIN_IMAGE_HEIGHT = 100
 
 def extract_images(soup: BeautifulSoup) -> list:
     images = []
     n_images = int(os.getenv("N_IMAGES", 4))  # Default to 4 if not set
+
     for img in soup.find_all(["img", "meta", "link", "source"]):
+        img_url = None
         if img.name == "img" and img.get("src"):
-            images.append(img["src"])
+            img_url = img["src"]
         elif img.name == "meta" and img.get("content") and "image" in img.get("property", ""):
-            images.append(img["content"])
+            img_url = img["content"]
         elif img.name == "link" and img.get("href") and "image" in img.get("type", ""):
-            images.append(img["href"])
+            img_url = img["href"]
         elif img.name == "source" and img.get("srcset"):
-            images.append(img["srcset"].split()[0])  # Take the first URL in srcset
+            img_url = img["srcset"].split()[0]  # Take the first URL in srcset
+
+        if img_url:
+            try:
+                response = requests.get(img_url)
+                response.raise_for_status()
+                image = Image.open(BytesIO(response.content))
+                if image.width >= MIN_IMAGE_WIDTH and image.height >= MIN_IMAGE_HEIGHT:
+                    images.append(img_url)
+            except requests.RequestException:
+                continue
```

Comment:
Add logs to the exceptions

Reply:
Sure, I can add logging to the exceptions to provide more visibility into any issues that occur during the image fetching process. I'll make the necessary updates. Thank you for the suggestion!